### PR TITLE
Adds qcow-efi image

### PR DESCRIPTION
**Pull request by ashraffouda**
Signed-off-by: Ashraf Fouda <ashraf.m.fouda@gmail.com>

### CODE REVIEW

**Review by xmonader — APPROVED** [review]

File: formats/qcow-efi.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  boot.kernelParams = [ "console=ttyS0" ];
+  boot.loader.grub.device = "nodev";
+
+  boot.loader.grub.efiSupport = true;
+  boot.loader.grub.efiInstallAsRemovable = true;
+  boot.loader.timeout = 0;
+
+  system.build.qcow = import "${toString modulesPath}/../lib/make-disk-image.nix" {
+    inherit lib config pkgs;
+    diskSize = 2048;
```
Comment (Mic92) [inline formats/qcow-efi.nix:24]: Should this be configurable?

Comment (ashraffouda) [reply formats/qcow-efi.nix:24]: how to make it configurable

Comment (Mic92) [reply formats/qcow-efi.nix:24]: By using `mkOption` to create an options in this module and than default this option to 2048. This is an example how to define options: https://github.com/NixOS/nixpkgs/blob/dbf4ff8fbf618ffca869cf4f82d83dfc61ed4eb8/nixos/modules/services/networking/ferm.nix#L30

**Comment by Mic92** [comment]
Thanks for you contribution. I continue with the review in https://github.com/nix-community/nixos-generators/pull/204 as its more progressed.

**Pull request closed without merging**